addons/mrp/tests/test_backorder.py

```diff
@@ -225,6 +225,53 @@ def test_tracking_backorder_series_lot_1(self):
         self.assertEqual(self.env['stock.quant']._get_available_quantity(p_final, self.stock_location, lot_id=lot_final), nb_product_todo, f'You should have the {nb_product_todo} final product in stock')
         self.assertEqual(len(production.procurement_group_id.mrp_production_ids), nb_product_todo)
 
+    def test_tracking_backorder_series_lot_2(self):
+        """ Create an MO with component  tracked by lots
+        Consume lots from component different from default sugestion
+        and check reserved lots for BO.
+        """
+        production, _, _, p1, p2 = self.generate_mo(tracking_base_2='lot')
+        self.env['stock.quant']._update_available_quantity(p1, self.stock_location, 20)
+        lot1, lot2 = self.env['stock.lot'].create([{
+            'name': f'lot_consumed_{i}',
+            'product_id': p2.id,
+            'company_id': self.env.company.id,
+        } for i in range(2)])
+        self.env['stock.quant']._update_available_quantity(p1, self.stock_location, 20)
+        self.env['stock.quant']._update_available_quantity(p2, self.stock_location, 3, lot_id=lot1)
+        self.env['stock.quant']._update_available_quantity(p2, self.stock_location, 2, lot_id=lot2)
+
+        production.action_assign()
+
+        production_form = Form(production)
+        production_form.qty_producing = 3
+
+        details_operation_form = Form(production.move_raw_ids.filtered(lambda m: m.product_id == p1), view=self.env.ref('stock.view_stock_move_operations'))
+        with details_operation_form.move_line_ids.edit(0) as ml:
+            ml.qty_done = 4 * 3
+        details_operation_form.save()
+
+        # Consume 1 Product from lot1 and 2 from lot 2
+        self.assertEqual(len(production.move_raw_ids.filtered(lambda m: m.product_id == p2).move_line_ids), 2, 'resrerved 3 products from lot1 and 2 from lot2')
+        details_operation_form = Form(production.move_raw_ids.filtered(lambda m: m.product_id == p2), view=self.env.ref('stock.view_stock_move_operations'))
+        with details_operation_form.move_line_ids.edit(0) as ml:
+            ml.qty_done = 1
+            ml.lot_id = lot1
+        with details_operation_form.move_line_ids.edit(1) as ml:
+            ml.qty_done = 2
+            ml.lot_id = lot2
+        details_operation_form.save()
+
+        production = production_form.save()
+        action = production.button_mark_done()
+        backorder = Form(self.env['mrp.production.backorder'].with_context(**action['context']))
+        backorder.save().action_backorder()
+
+        p2_bo_mls = production.procurement_group_id.mrp_production_ids[-1].move_raw_ids.filtered(lambda m: m.product_id == p2).move_line_ids
+        self.assertEqual(len(p2_bo_mls), 1, 'reserved 2 products from lot1')
+        self.assertEqual(p2_bo_mls.lot_id, lot1, 'reserved 2 products from lot1')
+        self.assertEqual(p2_bo_mls.reserved_qty, 2, 'reserved 2 products from lot1')
```

Comment:
Same comment as for the other PR, most of the time assert messages are useless, it should only be used when we need to explain the expected value
```suggestion
        self.assertEqual(len(p2_bo_mls), 1)
        self.assertEqual(p2_bo_mls.lot_id, lot1)
        self.assertEqual(p2_bo_mls.reserved_qty, 2)
```